openedx/openedx-platform
# [BD-03] [TNL-7349] [BB-2734] Studio UI to edit discussions toggle

This adds Studio UI to edit discussions toggle.

It contains changes from #24380, because GitHub doesn't allow changing repository for existing PRs. It will be rebased on `master` after merging the mentioned PR. **Only the last commit should be reviewed for now.**

**JIRA tickets**: TNL-7349

**Dependencies**: #24380

**Screenshots**:
![image](https://user-images.githubusercontent.com/3189670/88421502-f8e3b200-cde8-11ea-91ef-0db72793a25e.png)
![image](https://user-images.githubusercontent.com/3189670/88421529-013bed00-cde9-11ea-98c7-f136a88f4666.png)
![image](https://user-images.githubusercontent.com/3189670/88421565-1153cc80-cde9-11ea-9cad-2f24c232ff81.png)

**Merge deadline**: None, after #24380

**Testing instructions**:

1. Open the course via Studio.
1. Click on the "setttings" button of a unit, enable discussions there.
1. Refresh the page and see that the settings are being retained.
1. Click on the parent subsection's/section's settings and see that the indeterminate status is visible for the discussions there. This is because the discussions are enabled only on some of its children.
1. Check that the recursive enabling/disabling as well as doing nothing and saving works properly.

**Reviewers**
